INN-3033: make code editor support tailwind/system color schema and overrides

jacobheric opened the pull request:
## Description

Code editor was previously hard coded to use a dark theme. Now it will use the tailwind/system color scheme by default and also support the ability to override the color scheme per use. This is useful for things like the current function view in the dashboard that we still want dark despite the fact that the dashboard itself use a light color scheme.

Here the webhooks editor gets the default light:
<img width="1324" alt="Screenshot 2024-05-14 at 3 31 06 PM" src="https://github.com/inngest/inngest/assets/191780/25330a4f-4c5d-4e74-af44-4854a897956f">

And the function editor is overriden to dark
<img width="1162" alt="Screenshot 2024-05-14 at 12 47 51 PM" src="https://github.com/inngest/inngest/assets/191780/9cd9600d-0941-4bb3-999f-8e4a03f449ba">


## Motivation
<!--- Please edit this to include the reason why we are making this change. -->

## Type of change (choose one)
- [x ] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ x] I've linked any associated issues to this PR.
- [ x] I've tested my own changes.

https://linear.app/inngest/issue/INN-3033/add-light-theme-to-code-block

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

jacobheric commented:
There are a couple visual quirks to the editor that light mode exposed (see lower border corners). I'll clean those up.

jacobheric commented:
Do we want to make the editor toolbar light as well? I've left it dark for now as it more or less matches the rest of the current design.

anafilipadealmeida commented:
> Do we want to make the editor toolbar light as well? I've left it dark for now as it more or less matches the rest of the current design.

It's a great question. Yes! In light mode, the border of the entire block should be `border-slate-200`, we should replace the backgrounds by `bg-white`, and text and description `text-white` by `text-slate-700`. We can ignore the button colors for now

jacobheric commented:
I believe I've addressed all the items we talked about in review with the exception of the transparent scrollbars. I couldn't find any easy way to do that via monaco options so I just let that be for now. 

Updates are: 
1. use figma colors for all light theme attributes
2. update header to respect light theme
3. use outline buttons that support dark/light in header
4. remove manual dark override and instead update helper dark detector to find any parent level tailwind dark class
5. fixe broken border corners on all editors after the first

They now look like:
<img width="1725" alt="Screenshot 2024-05-15 at 2 19 37 PM" src="https://github.com/inngest/inngest/assets/191780/6b106e17-1a2c-4531-b923-ae176a0107d4">
<img width="1724" alt="Screenshot 2024-05-15 at 2 19 57 PM" src="https://github.com/inngest/inngest/assets/191780/3e9fd694-72c7-452d-b899-1e8612f14075">

jacobheric commented on ui/packages/components/src/CodeBlock/CodeBlock.tsx:
```diff
@@ -133,15 +196,6 @@ export function CodeBlock({ header, tabs, actions = [] }: CodeBlockProps) {
     setActiveTab(index);
   };
 
-  function handleEditorDidMount(editor: MonacoEditorType) {
```
This was the cause of the broken borders as it was not adding this class when there were more than one editor on the page. So instead I just removed it and applied this globally via css, see above.

anafilipadealmeida commented on ui/packages/components/src/CodeBlock/CodeBlock.tsx:
```diff
@@ -52,8 +149,10 @@ interface CodeBlockProps {
+  const [dark, setDark] = useState(isDark());
   const [activeTab, setActiveTab] = useState(0);
   const editorRef = useRef<MonacoEditorType>(null);
+  const wrapperRef = useRef<any>(null);
```
```suggestion
  const wrapperRef = useRef<HTMLDivElement>(null);
```

anafilipadealmeida commented on ui/packages/components/src/utils/theme.ts:
```diff
@@ -0,0 +1,12 @@
+// This will futureproof us a bit if we switch to manual or system color scheme.
+// If an element ref is provided we look for a dark class in any parents.
+// This is useful for components that might have dark theme sections inside a general light theme
+export const colorScheme = (elementRef?: any): 'light' | 'dark' =>
```
I think both colorScheme and isDark can have a `HTMLElement` type instead of `any`

jacobheric replied:
Yup, will do!

anafilipadealmeida commented on ui/packages/components/src/CodeBlock/CodeBlock.tsx:
```diff
@@ -66,62 +165,26 @@ export function CodeBlock({ header, tabs, actions = [] }: CodeBlockProps) {
 
+  useEffect(() => {
+    //
+    // We don't have a dom ref until we're renedered, so check for dark theme parent classes then
```
Nitpick: 
```suggestion
    // We don't have a DOM ref until we're rendered, so check for dark theme parent classes then
```

anafilipadealmeida reviewed (commented):
Looking good @jacobheric ! Left a few small comments 🙂

jacobheric commented:
Thanks @anafilipadealmeida for the careful review, I think I've addressed all your comments! Let me know if you see anything else.  

Also, I see I'm getting an invalid dom nesting error now that I've switched to the new buttons. I'll give that a look and fix before we merge.

anafilipadealmeida commented on ui/packages/components/src/CodeBlock/CodeBlock.tsx:
```diff
@@ -309,9 +367,9 @@ export function CodeBlock({ header, tabs, actions = [] }: CodeBlockProps) {
                         : 'block border-b outline-none transition-all duration-150',
-                      isActive && 'border-indigo-400 text-white',
+                      isActive && 'border-indigo-400 text-indigo-500 dark:text-white',
                       !isActive && !isSingleTab && 'border-transparent text-slate-400'
```
```suggestion
                      !isActive && !isSingleTab && 'border-transparent text-slate-700 dark:text-slate-400'
```

anafilipadealmeida reviewed (approved):
Thank you for all the hard work, looking great! Left a final small comment, but approving this already!
